plum/resolvable.py

```diff
@@ -41,6 +41,15 @@ def deliver(self, obj):
 
     def resolve(self):
         if self._obj is None:
-            raise ResolutionError("Promise was not kept.")
+            raise ResolutionError("Promise `{}` was not kept.".format(repr(self)))
         else:
             return self._obj
+
+    @property
+    def resolved(self):
+        """bool: True if the Promise has been kept."""
+        return self._obj is not None
+
+    def __repr__(self):
+        # fallback in case superclasses don't overload repr
+        return "Promise(obj=self._obj)"
```

Comment:
```suggestion
        # Fallback in case subclasses don't overload `__repr__`.
        return f"Promise(obj={self._obj!r})"
```

Should this also check whether the promise has already been resolved, because `self._obj` might not exist yet? I.e., maybe give `<Promise: unfulfilled>` back if it isn't yet resolved?

Reply:
that is a good idea, indeed. I implemented it